cli/logger.ts

```diff
@@ -51,9 +50,11 @@ export class Logger< T extends string > {
 		this.spinner.text = message;
 	}
 
-	public reportSuccess( message: string ) {
+	public reportSuccess( message: string, shouldClearSpinner = false ) {
```

Comment:
Nit: I think this logic could be a bit cleaner.
Could we maybe make the clear spinner decision based on the current action?
From what I could tell, all the cases for clear are on `LoggerAction.VALIDATE`

Reply:
I actually opted for this approach initially. In short, I added a `shouldClearSpinnerForAction` argument to the `Logger` constructor. This argument accepted a function that mapped `T` to a boolean. This approach worked fine, I just found it kinda contrived compared to the current approach. 

Happy to discuss this further and potentially adjust in a follow-up PR.